tests/test_mq_service.py

```diff
@@ -1362,3 +1363,24 @@ def test_owner_id_present_in_existing_host_but_missing_from_payload(mq_create_or
     assert updated_key == created_key
     assert updated_event["host"]["system_profile"]["owner_id"] == NEW_CN
     assert updated_event["host"]["display_name"] == "better_test_host"
+
+
+@pytest.mark.parametrize(
+    "user",
+    (
+        {"user": {"email": "tuser@redhat.com", "first_name": "test"}},
+        {"user": {"email": "tuser@redhat.com"}},
+        {"user": {"first_name": "test"}},
+        {"user": {}},
```

Comment:
These are good, though I think we should include a scenario for when the "user" field is straight-up missing, which is the scenario that prompted this PR in the first place:

```suggestion
        {"user": {"email": "tuser@redhat.com", "first_name": "test"}},
        {"user": {"email": "tuser@redhat.com"}},
        {"user": {"first_name": "test"}},
        {"user": {}},
        {},
```

Reply:
Done!